check_diff/src/lib.rs

```diff
@@ -270,3 +347,36 @@ pub fn compile_rustfmt(
         feature_runner,
     });
 }
+
+fn search_for_rs_files(repo: &Path) -> impl Iterator<Item = PathBuf> {
+    return WalkDir::new(repo)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|entry| {
+            let path = entry.path();
+            return path.is_file() && path.extension().map_or(false, |ext| ext == "rs");
+        })
+        .map(|entry| entry.into_path());
+}
+
+pub fn check_diff(config: Option<Vec<String>>, runners: CheckDiffRunners, repo: &Path) -> i32 {
+    let mut errors = 0;
+    search_for_rs_files(repo).for_each(|file| match runners.create_diff(file.as_path(), &config) {
+        Ok(diff) => {
+            if !diff.is_empty() {
+                eprint!("{diff}");
+                errors += 1;
+            }
+        }
+        Err(e) => {
+            eprintln!(
+                "Error creating diff for {:?}: {:?}",
+                file.as_path().display(),
+                e
+            );
+            errors += 1;
+        }
+    });
```

Comment:
Personal preference, but I really think the code is clearer when written using a `for loop`. Please update this when you get a chance.